package/src/lib/Assertion.ts

```diff
@@ -429,6 +440,43 @@ export class Assertion<T> {
     });
   }
 
+  /**
+   * Checks if the value is of a specific data type. The supported data types
+   * are the same as the `typeof` operator, plus an additional `array` which
+   * allows desabiguation between `object` (which can also be an array).
+   *
+   * @example
+   * ```
+   * const arr = [1, 2, 3];
+   *
+   * expect(arr).toBeOfType("array");
+   * expect(arr[0]).toBeOfType("number");
+   * expect(arr[9]).toBeOfType("undefined");
+   * ```
+   *
+   * @param expected the expected data type
+   * @returns the assertion instance
+   */
+  public toBeOfType(expected: DataType): this {
```

Comment:
I'm not overly fond of the name of this method, so I'm open to suggestions 🙂

Reply:
I think it's fine. It reads naturally when used and it's pretty self explanatory. :)